ipatests/test_integration/test_ipahealthcheck.py

```diff
@@ -904,3 +910,101 @@ def test_ipahealthcheck_trust_agent_member_check(self):
         for check in data:
             assert check["result"] == "SUCCESS"
             assert check["kw"]["key"] == self.master.hostname
+
+
+class TestIpaHealthCheckWithExternalCA(IntegrationTest):
+    """
+    Tests to run and check whether ipa-healthcheck tool
+    reports correct status when IPA server is configured
+    with external CA.
+    """
+
+    @classmethod
+    def install(cls, mh):
+        result = install_server_external_ca_step1(cls.master)
+        assert result.returncode == 0
+        root_ca_fname, ipa_ca_fname = tasks.sign_ca_and_transport(
+            cls.master, paths.ROOT_IPA_CSR, ROOT_CA, IPA_CA
+        )
+
+        install_server_external_ca_step2(
+            cls.master, ipa_ca_fname, root_ca_fname
+        )
+        tasks.kinit_admin(cls.master)
+
+    def test_ipahealthcheck_crlmanagercheck(self):
+        """
+        Test for IPACRLManagerCheck
+        """
+        returncode, data = run_healthcheck(
+            self.master, "ipahealthcheck.ipa.roles", "IPACRLManagerCheck"
+        )
+        assert returncode == 0
+        for check in data:
+            assert check["result"] == "SUCCESS"
+            assert check["kw"]["key"] == "crl_manager"
+            assert check["kw"]["crlgen_enabled"] is True
+
+    def test_ipahealthcheck_certmongerca(self):
+        """
+        Test for IPACertmongerCA
+        """
+        returncode, data = run_healthcheck(
+            self.master, "ipahealthcheck.ipa.certs", "IPACertmongerCA",
+        )
+        assert returncode == 0
+        for check in data:
+            assert check["result"] == "SUCCESS"
+            assert (
+                check["kw"]["key"] == "IPA"
+                or check["kw"]["key"] == "dogtag-ipa-ca-renew-agent"
+                or check["kw"]["key"] == "dogtag-ipa-ca-renew-agent-reuse"
+            )
```

Comment:
This will fail to catch if for some crazy reason ipa-healthcheck doesn't return one of the values. You might add a counter to ensure that all 3 are picked up.

Reply:
And moreso if you want to delete one or more of the CA's with getcert remove-ca -c CA_NAME and re-run healthcheck to confirm it is gone that would be even better. With a fixture to re-add.

Reply:
Removed the existing test and tried deleting the 'dogtag-ipa-ca-renew-agent' cert using getcert and an error msg is displayed 'Certmonger CA '{key}' missing'